pkg/controller/controller_instance.go

```diff
@@ -302,6 +333,33 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 	glog.V(4).Infof("Provisioning a new Instance %v/%v of ServiceClass %v at Broker %v", instance.Namespace, instance.Name, serviceClass.Name, brokerName)
 	response, err := brokerClient.ProvisionInstance(request)
 	if err != nil {
+		// There are two buckets of errors to handle:
+		// 1.  Errors that represent a failure response from the broker
+		// 2.  All other errors
+		if osb.IsHTTPError(err) {
```

Comment:
for the future, it'd be nice to have this return the `osb.HTTPStatusCodeError` so you don't have to type-assert it below.

I've created https://github.com/pmorie/go-open-service-broker-client/issues/47 to track that